federation/pkg/kubefed/util/util.go

```diff
@@ -254,9 +254,9 @@ func GetVersionedClientForRBACOrFail(hostFactory cmdutil.Factory) (client.Interf
 				}
 				return hostFactory.ClientSetForVersion(&gv)
 			}
-			for i := 0; i < len(g.Versions); i++ {
-				if g.Versions[i].GroupVersion != "" {
-					gv, err := schema.ParseGroupVersion(g.Versions[i].GroupVersion)
+			for _, version := range g.Versions {
+				if version.GroupVersion != "" {
+					gv, err := schema.ParseGroupVersion(version.GroupVersion)
```

Comment:
In function the original point equals the modification point. However modification is much cleaner and lighter than the original point. First, it does not need len(xxx) to know the loop count, and the index is not necessary for the function. Don't you think so?